internal/provider_fm/resource_id.go

```diff
@@ -0,0 +1,219 @@
+package provider_fm
+
+import (
+	"context"
+	"crypto/rand"
+	"encoding/base64"
+	hex2 "encoding/hex"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"math/big"
+	"strings"
+)
+
+type resourceIDType struct{}
+
+func (r resourceIDType) GetSchema(context.Context) (tfsdk.Schema, diag.Diagnostics) {
+	return tfsdk.Schema{
+		Description: `
+The resource ` + "`random_id`" + ` generates random numbers that are intended to be
+used as unique identifiers for other resources.
+
+This resource *does* use a cryptographic random number generator in order
+to minimize the chance of collisions, making the results of this resource
+when a 16-byte identifier is requested of equivalent uniqueness to a
+type-4 UUID.
+
+This resource can be used in conjunction with resources that have
+the ` + "`create_before_destroy`" + ` lifecycle flag set to avoid conflicts with
+unique names during the brief period where both the old and new resources
+exist concurrently.
+`,
+		Attributes: map[string]tfsdk.Attribute{
+			"keepers": {
+				Description: "Arbitrary map of values that, when changed, will trigger recreation of " +
+					"resource. See [the main provider documentation](../index.html) for more information.",
+				Type: types.MapType{
+					ElemType: types.StringType,
+				},
+				Optional:      true,
+				PlanModifiers: []tfsdk.AttributePlanModifier{tfsdk.RequiresReplace()},
+			},
+			"byte_length": {
+				Description: "The number of random bytes to produce. The minimum value is 1, which produces " +
+					"eight bits of randomness.",
+				Type:          types.Int64Type,
+				Required:      true,
+				PlanModifiers: []tfsdk.AttributePlanModifier{tfsdk.RequiresReplace()},
+			},
+			"prefix": {
+				Description: "Arbitrary string to prefix the output value with. This string is supplied as-is, " +
+					"meaning it is not guaranteed to be URL-safe or base64 encoded.",
+				Type:          types.StringType,
+				Optional:      true,
+				PlanModifiers: []tfsdk.AttributePlanModifier{tfsdk.RequiresReplace()},
+			},
+			"b64_url": {
+				Description: "The generated id presented in base64, using the URL-friendly character set: " +
+					"case-sensitive letters, digits and the characters `_` and `-`.",
+				Type:     types.StringType,
+				Computed: true,
+			},
+			"b64_std": {
+				Description: "The generated id presented in base64 without additional transformations.",
+				Type:        types.StringType,
+				Computed:    true,
+			},
+			"hex": {
+				Description: "The generated id presented in padded hexadecimal digits. This result will " +
+					"always be twice as long as the requested byte length.",
+				Type:     types.StringType,
+				Computed: true,
+			},
+			"dec": {
+				Description: "The generated id presented in non-padded decimal digits.",
+				Type:        types.StringType,
+				Computed:    true,
+			},
+			"id": {
```

Comment:
Once `v0.8.0` is shipped (https://github.com/hashicorp/terraform-plugin-docs/pull/134), the docs tools won't need us to set explicitly the `id` in the schema anymore (but it will still be possible).

Reply:
Nice! Will look out for that.